stac_pydantic/extensions.py

```diff
@@ -1,18 +1,22 @@
-from typing import Union
+import json
+from typing import Any, Dict, Union
 
 import jsonschema
 import requests
 
-from stac_pydantic import Catalog, Collection, Item
+from stac_pydantic.catalog import Catalog
+from stac_pydantic.collection import Collection
+from stac_pydantic.item import Item
 
 
 def validate_extensions(
-    stac_obj: Union[Item, Collection, Catalog, dict], reraise_exception: bool = False
+    stac_obj: Union[Item, Collection, Catalog, Dict[str, Any]],
+    reraise_exception: bool = False,
 ) -> bool:
     if isinstance(stac_obj, dict):
         stac_dict = stac_obj
     else:
-        stac_dict = stac_obj.dict()
+        stac_dict = json.loads(stac_obj.model_dump_json())
```

Comment:
why not using `stac_obj.model_dump()`?

Reply:
Yes, I agree this looks odd. I tried this first. But `model_dump` returns Python types like `URL` and `datetime`.  `jsonschema` doesn't like this and expects the string representation.
I will add an inline comment.